server/src/main/java/org/opensearch/common/settings/WriteableSetting.java

```diff
@@ -147,33 +162,107 @@ private Setting<?> createSetting(
                     ? Setting.boolSetting(key, (boolean) defaultValue, propertyArray)
                     : Setting.boolSetting(key, (Setting<Boolean>) fallback.getSetting(), propertyArray);
             case Integer:
-                return fallback == null
-                    ? Setting.intSetting(key, (int) defaultValue, propertyArray)
-                    : Setting.intSetting(key, (Setting<Integer>) fallback.getSetting(), propertyArray);
+                if (fallback == null) {
```

Comment:
This is more of a nit pick - but the code would be cleaner if the statements were:

```
if (parser instanceof Writeable && fallback==null) {
...
} else if (fallback==null) {
...
} else {
...
}
```

Reply:
ACK